nomic/atlas.py

```diff
@@ -2,1315 +2,234 @@
 or in a Jupyter Notebook to organize and interact with your unstructured data.
 """
-import os
-import pickle
-import concurrent.futures
-import json
-import uuid
-import gc
-import io
-import time
-import base64
+
 from typing import Dict, List, Optional
-from uuid import UUID
 
 import numpy as np
-import requests
 from loguru import logger
-from pydantic import BaseModel, Field
 from tqdm import tqdm
-from datetime import date
 
+from .project import AtlasProject
+from .settings import *
 from .utils import get_random_name
-from .cli import get_api_credentials, refresh_bearer_token, validate_api_http_response
-import sys
-# Uploads send several requests to allow for threadpool refreshing.
-# Threadpool hogs memory and new ones need to be created.
-# This number specifies how much data gets processed before a new Threadpool is created
-MAX_MEMORY_CHUNK = 150000
-EMBEDDING_PAGINATION_LIMIT = 1000
-ATLAS_DEFAULT_ID_FIELD = 'id_'
-DEFAULT_PROJECTION_N_NEIGHBORS = 15
-DEFAULT_PROJECTION_EPOCHS = 50
-DEFAULT_PROJECTION_SPREAD = 1.
-
-def get_object_size_in_bytes(obj):
-    marked = {id(obj)}
-    obj_q = [obj]
-    sz = 0
-
-    while obj_q:
-        sz += sum(map(sys.getsizeof, obj_q))
-
-        # Lookup all the object referred to by the object in obj_q.
-        # See: https://docs.python.org/3.7/library/gc.html#gc.get_referents
-        all_refr = ((id(o), o) for o in gc.get_referents(*obj_q))
 
-        # Filter object that are already marked.
-        # Using dict notation will prevent repeated objects.
-        new_refr = {o_id: o for o_id, o in all_refr if o_id not in marked and not isinstance(o, type)}
 
-        # The new obj_q will be the ones that were not marked,
-        # and we will update marked with their ids so we will
-        # not traverse them again.
-        obj_q = new_refr.values()
-        marked.update(new_refr.keys())
-
-    return sz
-
-def assert_valid_project_id(project_id):
-    try:
-        uuid_obj = UUID(project_id, version=4)
-    except ValueError:
-        raise ValueError(f"`{project_id}` is not a valid project id.")
-
-class CreateIndexResponse(BaseModel):
-    map: Optional[str] = Field(
-        None, description="A link to the map this index creates. May take some time to be ready so check the job state."
+def map_embeddings(
+    embeddings: np.array,
+    data: List[Dict] = None,
+    id_field: str = None,
+    map_name: str = None,
+    map_description: str = None,
+    is_public: bool = True,
+    colorable_fields: list = [],
+    build_topic_model: bool = False,
+    topic_label_field: str = None,
+    num_workers: int = 10,
+    organization_name: str = None,
+    reset_project_if_exists: bool = False,
+    add_datums_if_exists: bool = False,
+    shard_size: int = 1000,
+    projection_n_neighbors: int = DEFAULT_PROJECTION_N_NEIGHBORS,
+    projection_epochs: int = DEFAULT_PROJECTION_EPOCHS,
+    projection_spread: float = DEFAULT_PROJECTION_SPREAD,
+) -> AtlasProject:
```

Comment:
several params are not documented (projection_n_neighbors, projection_epochs, etc...)